skipruntime-ts/core/src/api.ts

```diff
@@ -84,14 +84,16 @@ export interface Reducer<V extends Json, A extends Json> {
 }
 
 /**
- * An iterable collection of dependency-safe values.
+ * A non-empty iterable collection of dependency-safe values.
  */
 export interface Values<T> extends Iterable<T & DepSafe> {
   /**
    * Return the first value, if there is exactly one.
-   * @throws {@link SkipNonUniqueValueError} if this iterable contains either zero or multiple values.
+   * @param options - Optional named parameters
+   * @param options.default - (optional) A default value to return if this contains multiple values.
+   * @throws {@link SkipNonUniqueValueError} if this iterable contains multiple values and no default is provided.
    */
-  getUnique(): T & DepSafe;
+  getUnique(options?: { default: T }): T & DepSafe;
```

Comment:
```suggestion
  getUnique(options?: { default: T & DepSafe }): T & DepSafe;
```
Here and elsewhere, it is probably better to require users to deepFreeze their own objects. The rest of the API requires the user to provided already-frozen values where needed rather than the framework mutating them on the users' behalf.

Reply:
I did it that way at first, but the syntax gets very clunky very fast... especially if we're also making defaults non-optional

Reply:
e.g. from one of our `./skipruntime-ts/examples`:
```diff
   mapEntry(gid: GroupID, group: Values<Group>): Iterable<[GroupID, UserID]> {
     return group
-      .getUnique()
+      .getUnique({
+        default: deepFreeze({ name: "Unknown Group", members: [] }),
+      })
       .members.flatMap((uid) =>
-        this.users.getUnique(uid).active ? [[gid, uid]] : [],
+        this.users.getUnique(uid, {
+          default: deepFreeze({ name: "Unknown User", friends: [] }),
+        }).active
+          ? [[gid, uid]]
+          : [],
       );
   }
 }
 ```

Reply:
Not pushing, but hmm: Even without deepFreeze, the default objects ought ideally to be let-bound before the return statement (as a field of the Mapper class even), no? Otherwise they have to be allocated each time the closure given to flatMap is called.

Reply:
Yes, that's what I actually have in the code (also since they are needed ~5 times throughout that example).  Copied like this to make it self-contained. Still working on pushing required defaults through all of the examples, I think it's a mixed bag so far.